src/main/java/com/squareup/javapoet/JavaFile.java

```diff
@@ -94,7 +102,8 @@ public void writeTo(Path directory) throws IOException {
     }
 
     Path outputPath = outputDirectory.resolve(typeSpec.name + ".java");
-    try (Writer writer = new OutputStreamWriter(Files.newOutputStream(outputPath))) {
+    try (Writer writer = Files.newBufferedWriter(outputPath,
+            charset != null ? charset : Charset.defaultCharset())) {
```

Comment:
Personal opinion: I'd rather have it throw if `null`, and have the overload explicitly pass the default charset in.